posts/Modernize-your-pipeline-to-be-ARM-ready-with-multi-arch-container-builds.md

```diff
@@ -0,0 +1,233 @@
+
+## Building a multi-arch image and deploying to the EKS cluster
+
+For this section, we will be building a tiny python program. This program, when it runs, prints out the architecture of the machine it is running on. After we code up the program, we will build it inside a gitlab pipeline using docker's buildx. Once built, it will be pushed to a docker registry. Finally, we will deploy it to our kubernetes cluster and see the architecture print out. Please note, additional networking/security steps may be required in order for EKS to pull from a specified container registry.  It's okay if you can't follow along for every step, the important part is understanding the bigger picture. 
+
+In your favorite code editor, write the python program into a file named `print-platform-info.py`:
+```python
+import platform
+print(platform.uname())    
+```
+
+Next, create a `Dockerfile` in the same directory as your python program
+
+```Dockerfile
+FROM python
+COPY ./print-platform-info.py /print-platform-info
+CMD ["python3", "/print-platform-info"]
+```
+Now this is typically where you would scamper off to the command line and run `docker build -t my_image .`, but hold off for just a minute! We have to set docker to use a different *build instance*. In the command line inside the project directory where you have your `Dockerfile` run:
+```bash
+docker buildx create --use
+```
+You may need to add some additional arguments to authenticate to your docker registy in the above command. Check the [docs](https://docs.docker.com/engine/reference/commandline/buildx_use/) for any information or google any errors you may run into. Now it is time to run the actual build command. Back at the command prompt, run:
+```bash
+`docker buildx build --platform linux/arm64,linux/amd64 -t <registry-url>/<image-name>:<image-tag> . --push`
+```
+Let's take a look at the different pieces of this command...
+* `docker buildx build` - This simply tells docker to use the buildx build instance that we set before.
+* `--platform linux/arm64,linux/amd64` - tells buildx to produce an image for arm64 and amd64. The builder instance actually pulls down the native architecture image to build on.
+* `-t <registry-url>/<iamge-name>:<image-tag>'` - This is where you specify your registry url and give your image a name and a tag. If you don't specify a registry, one of your images may end up living in the docker cache.
+* `. --push` - tells docker to use the local directory Dockerfile and then push the built images to our registry.
```

Comment:
. has to do with the directory that is tar and sent to the docker daemon, which in turn if there is a "Dockerfile" in that directory it will get picked up automatically.